braindecode/models/modules.py

```diff
@@ -523,3 +529,331 @@ def forward(self, x):
+    It uses MNE's `create_filter` function to create the band-specific filters and
+    applies them to multi-channel time-series data. Each filter in the bank corresponds to a
+    specific frequency band and is applied to all channels of the input data. The filtering is
+    performed using FFT-based convolution via the `fftconvolve` function from
+    :func:`torchaudio.functional if the method is FIR, and `filtfilt` function from
+    :func:`torchaudio.functional if the method is IIR.
+
+    The default configuration creates 9 non-overlapping frequency bands with a 4 Hz bandwidth,
+    spanning from 4 Hz to 40 Hz (i.e., 4-8 Hz, 8-12 Hz, ..., 36-40 Hz). This setup is based on the
+    reference: *FBCNet: A Multi-view Convolutional Neural Network for Brain-Computer Interface*.
+
+    Parameters
+    ----------
+    n_chans : int
+        Number of channels in the input signal.
+    sfreq : int
+        Sampling frequency of the input signal in Hz.
+    band_filters : Optional[List[Tuple[float, float]]] or int, default=None
+        List of frequency bands as (low_freq, high_freq) tuples. Each tuple defines
+        the frequency range for one filter in the bank. If not provided, defaults
+        to 9 non-overlapping bands with 4 Hz bandwidths spanning from 4 to 40 Hz.
+    filter_length : str | int
+        Length of the FIR filter to use (if applicable):
+
+        * **'auto' (default)**: The filter length is chosen based
+          on the size of the transition regions (6.6 times the reciprocal
+          of the shortest transition band for fir_window='hamming'
+          and fir_design="firwin2", and half that for "firwin").
+        * **str**: A human-readable time in
+          units of "s" or "ms" (e.g., "10s" or "5500ms") will be
+          converted to that number of samples if ``phase="zero"``, or
+          the shortest power-of-two length at least that duration for
+          ``phase="zero-double"``.
+        * **int**: Specified length in samples. For fir_design="firwin",
+          this should not be used.
+    l_trans_bandwidth : Union[str, float, int], default='auto'
+        Width of the transition band at the low cut-off frequency in Hz
+        (high pass or cutoff 1 in bandpass). Can be "auto"
+        (default) to use a multiple of ``l_freq``::
+
+            min(max(l_freq * 0.25, 2), l_freq)
+
+        Only used for ``method='fir'``.
+    h_trans_bandwidth : Union[str, float, int], default='auto'
+        Width of the transition band at the high cut-off frequency in Hz
+        (low pass or cutoff 2 in bandpass). Can be "auto"
+        (default in 0.14) to use a multiple of ``h_freq``::
+
+            min(max(h_freq * 0.25, 2.), info['sfreq'] / 2. - h_freq)
+
+        Only used for ``method='fir'``.
+    method : str, default='fir'
+        ``'fir'`` will use FIR filtering, ``'iir'`` will use IIR
+        forward-backward filtering (via :func:`~scipy.signal.filtfilt`).
```

Comment:
Move this one before the method specific arguments to make this easier to read (so before `filter_length`).

Reply:
Maybe also add a link toward why to chose vs the other if you have one?